httpcore5-testing/src/test/java/org/apache/hc/core5/testing/nio/H2ProtocolNegotiationTest.java

```diff
@@ -233,7 +233,7 @@ public void testNegotiateProtocol() throws Exception {
         final HttpResponse response1 = message1.getHead();
         Assert.assertThat(response1.getCode(), CoreMatchers.equalTo(HttpStatus.SC_OK));
 
-        if (javaVersion >= 9) {
+        if (javaVersion >= 8) {
```

Comment:
I'm not sure what versions the various CI servers use, so it may be helpful to check for 8u252+

Reply:
Ancient on Trusty: java version "1.8.0_151"

Reply:
We don't have code which exposes the patch version. Since neither Jekins not Travis have recent JDKs installed, I don't see a way to move this forward :-(

Reply:
Looks like the backport was 8u251, not 8u252. off by one error in my previous comment: https://www.oracle.com/java/technologies/javase/8u251-relnotes.html

How about something like this?

```java
    private boolean isAlpnAvailable() {
        if (javaVersion > 8) {
            return true;
        }
        if (javaVersion == 8) {
            // ALPN was backported to java 8u251. Using a java 8 runtime this can be detected
            // based on the presence of the 'getApplicationProtocol' method.
            try {
                SSLEngine.class.getMethod("getApplicationProtocol");
                return true;
            } catch (NoSuchMethodException e) {
                return false;
            }
        }
        return false;
    }
```

Reply:
@michael-o This is the reason I have not enabled ALPN tests on Java 1.8. I could not find a reliable way of finding out the exact build number to make the test dependent on it. It is not worth it. Please close the PR.